pkg/aws/services/tagging_test.go

```diff
@@ -0,0 +1,421 @@
+package services
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+	gomock "github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestContainsTags(t *testing.T) {
+	tests := []struct {
+		contains   bool
+		findTags   Tags
+		sourceTags Tags
+		testName   string
+	}{
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			testName: "exact match with one tag",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			testName: "exact match with multiple tags",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			testName: "match one tag from source's multiple tags",
+		},
+		{
+			contains: true,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value1"),
+				"Key2": aws.String("Value2"),
+				"Key3": aws.String("Value3"),
+			},
+			testName: "match subnet of tags from source's multiple tags",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key2": aws.String("Value2"),
+			},
+			testName: "no match for tag with different key and value",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("Value2"),
+			},
+			testName: "no match for tag with same key, different value",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"Key1": aws.String("value1"),
+			},
+			testName: "no match for tag with same key, different value case",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value1"),
+			},
+			sourceTags: Tags{
+				"key1": aws.String("Value1"),
+			},
+			testName: "no match for tag with different key case",
+		},
+		{
+			contains: false,
+			findTags: Tags{
+				"Key1": aws.String("Value"),
+			},
+			sourceTags: Tags{},
+			testName:   "no match for tag when source has no tags",
```

Comment:
Might also be good to include a case where "findTags" is empty.

Reply:
Done. Had to change the `containsTags` function to cover this use case.